unit tests for syncRolloutStatus in pkg/controller/deployment

nulltrope opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:
Adds unit tests for syncRolloutStatus func in controller/deployment package.

**Which issue this PR fixes**: 
fixes #53663 

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```

Commit:
pkg/controller/deployment: unit tests for syncRolloutStatus

Created unit tests for syncRolloutStatus function in the deployment package.
Tests for syncRolloutStatus have brought the overall test coverage of deployment/progress to 82.1%.

Signed-off-by: John Kelly <jekohk@gmail.com>

jekohk commented:
/cc @kargakis

kargakis commented:
/ok-to-test

jekohk commented:
/assign @kargakis

kargakis commented on pkg/controller/deployment/progress_test.go:
```diff
@@ -161,3 +172,162 @@ func TestRequeueStuckDeployment(t *testing.T) {
+		},
+	}
+
+	fake := fake.Clientset{}
```
Move this inside the test loop since it's mutated every time an API action happens.

kargakis commented:
Can you also add a test case that exercises https://github.com/kubernetes/kubernetes/pull/53614?

Commit:
pkg/controller/deployment: syncRolloutStatus additional unit test case

Added additional test case that exercises fix #53614 or npe when scaling --replicas=0
Moved creation of clientset and deployment controller objects inside test loop since it's mutated every time an API action happens.

Signed-off-by: John Kelly <jekohk@gmail.com>

jekohk commented:
/test pull-kubernetes-e2e-kops-aws

jekohk commented:
/cc @dims

kargakis commented:
/lgtm
/approve
